### ISSUE TITLE
Babel config transpiles array ranges differently

### ISSUE THREAD

**Issue by oleblaesing** [storybookjs/storybook#12381]
**Describe the bug**
When I try to create a dynamic array range with this code inside my Storybook:

```js
[...Array(24).keys()]
```

I get the following output:

```
Array [ Array Iterator ]
```

**To Reproduce**

1. Create a new story
2. Put the following code into the new component
3. `console.log([...Array(24).keys()])`
4. Open the story
5. You should see the "broken" array example above

**Expected behavior**
The Babel config of my project transpiles that into correct code, that creates a regular array like I would expect. You can test it, when you put the code into your ES6 compatible browsers console:

```
Array [ 0, 1, 2, 3, 4, ... ]
```

**System:**
```
Environment Info:

  System:
    OS: macOS 10.15.6
    CPU: (4) x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
  Binaries:
    Node: 12.18.3 - /usr/local/opt/node@12/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.6 - /usr/local/opt/node@12/bin/npm
  Browsers:
    Chrome: 85.0.4183.83
    Edge: 85.0.564.44
    Firefox: 80.0.1
    Safari: 13.1.2
  npmPackages:
    @storybook/addon-knobs: ^5.3.0 => 5.3.18 
    @storybook/addon-storyshots: ^5.3.0 => 5.3.18 
    @storybook/addon-viewport: ^5.3.0 => 5.3.18 
    @storybook/addons: ^5.3.0 => 5.3.18 
    @storybook/react: ^5.3.0 => 5.3.18 
```

**Additional context**
Firefox is the browser I use.

**Comment by oleblaesing** [comment]
For now I can mitigate this issue by using this code, but it shouldn't be necessary at all:

```js
Array.from([...Array(24).keys()][0])
```

**Comment by shilman** [comment]
@oleblaesing I'm assuming this the out-of-box babel/webpack config with no customization?

**Comment by oleblaesing** [comment]
Yup, the only thing we customized is using a different loader for image files.

**Comment by ndelangen** [comment]
Fixed in 7.0 beta
